### DISCUSSION TITLE
How to send Events to and from different part of the application

### DISCUSSION THREAD

**Discussion by cerobe** [AvaloniaUI/Avalonia#14799] [Q&A] [answered]
I'm trying to implement a log window, the logs will come from different part of the application. 
My scenario is: I click a button and it will launch a different thread (in a service class) for a long task. I would like this thread to send event and log then in the window.

I looked at https://docs.avaloniaui.net/docs/concepts/input/routed-events but it seems to be events routing within the different views.

Could guide me in the correct direction?

**Comment by thevortexcloud** [comment]
Is there a reason you can't do this within your VM? You will need to update your UI with a dispatcher call. I have done this myself by just having the VM listen to a log event on the log sink so it knows when to update the UI.

**Comment by msneijders** [comment, marked as answer]
You should probably not look for an Avalonia solution for this, as things like this are out of its scope; but for example [CommunityToolkit.Messenger](https://learn.microsoft.com/en-us/dotnet/communitytoolkit/mvvm/messenger) or other messaging/event solutions.
